src/actions/actions.ts

```diff
@@ -1652,6 +1652,114 @@ class MoveLineBegin extends BaseMovement {
 }
 
 @RegisterAction
+class MoveScreenLineBegin extends BaseMovement {
+  modes = [ModeName.Normal, ModeName.Visual, ModeName.VisualLine];
+  keys = ["g", "0"];
+  viewPosition = "wrappedLineStart";
+  /**
+   * This paramter will be consumed only when `to` is `lineUp` or `lineDown`.
+   * For other screen line movements, we are always operating on the same screen line.
+   * So we make its default value as 0.
+   */
+  noOfLines = 0;
+
+  public async execAction(position: Position, vimState: VimState): Promise<Position | IMovement> {
+    await vscode.commands.executeCommand("cursorMove", {
+      to: this.viewPosition,
+      inSelectionMode: vimState.currentMode !== ModeName.Normal,
+      noOfLines: this.noOfLines
+    });
+
+    if (vimState.currentMode === ModeName.Normal) {
+      return Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.active);
+    } else {
+      /**
```

Comment:
I'm sure we can make one 😉 

Upon reflection, a mild slowdown here is acceptable, because not too many people use these commands (they are a bit esoteric). At this point, I think that a history API is priority number 1, and that we can safely bump the priority of getting actual positions down.

Right now the thing I would kill for is the history API. We are really suffering a lot because of that.